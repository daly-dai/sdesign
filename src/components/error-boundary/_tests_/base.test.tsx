import { render } from '@testing-library/react';
import { Button } from 'antd';
import React from 'react';
import { describe, expect, it } from 'vitest';

import SErrorBoundary from '../index';

import mountTest from '@dalydb/sdesign/tests/mountTest';

describe('error-boundary', () => {
  mountTest(SErrorBoundary);
  mountTest(() => <SErrorBoundary />);

  it('should render current', () => {
    const { container } = render(
      <SErrorBoundary>
        <Button type="primary">取消</Button>
        <Button type="primary">提交</Button>
      </SErrorBoundary>,
    );

    expect(container).toMatchSnapshot();
  });
});
