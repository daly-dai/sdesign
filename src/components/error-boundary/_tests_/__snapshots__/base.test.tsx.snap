// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`error-boundary > should render current 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-primary"
    type="button"
  >
    <span>
      取 消
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-primary"
    type="button"
  >
    <span>
      提 交
    </span>
  </button>
</div>
`;

exports[`error-boundary should render current 1`] = `
<div>
  <button
    class="ant-btn css-dev-only-do-not-override-1r287do ant-btn-primary"
    type="button"
  >
    <span>
      取 消
    </span>
  </button>
  <button
    class="ant-btn css-dev-only-do-not-override-1r287do ant-btn-primary"
    type="button"
  >
    <span>
      提 交
    </span>
  </button>
</div>
`;
